opendevin/runtime/runtime.py

```diff
@@ -29,13 +29,22 @@
 from opendevin.runtime.plugins import PluginRequirement
 
 
-def create_sandbox(sid: str = 'default', sandbox_type: str = 'exec') -> Sandbox:
+def create_sandbox(
+    sid: str = 'default',
+    sandbox_type: str = 'exec',
+    workspace: str | None = None,
+) -> Sandbox:
+    workspace = workspace or config.workspace_base
     if sandbox_type == 'exec':
-        return DockerExecBox(sid=sid, timeout=config.sandbox_timeout)
+        return DockerExecBox(
+            sid=sid, timeout=config.sandbox_timeout, workspace_mount_path=workspace
+        )
     elif sandbox_type == 'local':
-        return LocalBox(timeout=config.sandbox_timeout)
+        return LocalBox(timeout=config.sandbox_timeout, workspace=workspace)
     elif sandbox_type == 'ssh':
-        return DockerSSHBox(sid=sid, timeout=config.sandbox_timeout)
+        return DockerSSHBox(
+            sid=sid, timeout=config.sandbox_timeout, workspace_mount_path=workspace
+        )
```

Comment:
I think we should call the input parameter here the same thing, since these are all Sandbox objects. Maybe `relative_workspace`?

Then the mount path would be `path.join(config.workspace_mount_path, relative_workspace)`

I think it's also a good idea to keep the extra workspace directory param you're adding relative, rather than an absolute path. The base path varies between in-docker and out-of-docker, so it'd be nice to have a consistent reference to a subdirectory that's valid in both places

Reply:
Good point - done!